pkg/jsonnet/implementation/types/types.go

```diff
@@ -0,0 +1,10 @@
+package types
+
+type JsonnetVM interface {
+	EvaluateAnonymousSnippet(filename, snippet string) (string, error)
```

Comment:
It doesn't seem like it's really necessary for this method to know the filename since it's "anonymous" - this is something that go-jsonnet uses to show you nicer errors. I checked the binary implementation from #915 and [it ignores this too](https://github.com/grafana/tanka/blob/57ff948f60b8a399419d3f6905831a98d11307d2/pkg/jsonnet/implementation/binary/impl.go#L15-L19).

Would suggest this type:

```suggestion
	EvaluateAnonymousSnippet(snippet string) (string, error)
```

And arranging for the Go implementation to have the filename as a member on the wrapper struct.

Reply:
Filename can't be on the wrapper. The same evaluator could be used to eval multiple snippets of different files

Reply:
I'd leave it there

Reply:
Where in the code can that happen?

Could definitely be missing something but the only caller I can find is [this one](https://github.com/grafana/tanka/blob/c2bd64ede51b619b601430532b2164184ebcdd09/pkg/jsonnet/eval.go#L100) which calls `evaluateSnippet`, also with the `path`, and that always calls `implementation.Get` which returns a new instance, not a singleton, and then we call `MakeVM` on that to make a new VM.

Reply:
I think the "how it's being used currently" is not entirely relevant. A Go Jsonnet VM can be re-used if the import paths or ext code do not change. So, for our concerns, a VM/Evaluator is defined by those parameters. If we add the path to the `MakeEvaluator` command, then the interface gets weird

Also, I'm thinking I may add the filename to the anonymous eval error message for the binary implementation as well

Reply:
I would move the selection of the implementation outwards a bit, and then it becomes more natural to pass the path when you select the Go one.

See 82ae5aea80651854a97f264e332f3cf83fe8e24d for something like what I mean, in case that's clearer.

Reply:
Sure, that works for me. The only thing that is weird is that `vm.path` is ignored when you `EvaluateFile`, so the path does nothing in that case and you have to specify it twice. But it's not any weirder than what I was doing

Also, I like that the loaders will use the specified implementation for all operations, including listing envs

Can you merge your PR into this one, or make yours ready to review and we can close this one?

Reply:
Alright, cheers. I pushed that commit here, I'll give a :heavy_check_mark: for your bits, guess we can consider this a cross review heh